### DISCUSSION TITLE
[sitecore-jss-angular] Support standalone application

### DISCUSSION THREAD

**Discussion by spike1292** [Sitecore/jss#1818] [Ideas]
### Is your suggestion related to a problem? Please describe.

We want to use Angular standalone application. Currently this setup is not possible if you are using the sitecore jss angular library.

### Describe the solution you'd like

Support the angular modules and angular standalone way like angular router and forms do.

https://angular.dev/reference/migrations/standalone

### Additional information

https://angular.dev/guide/ngmodules/lazy-loading#standalone-application
https://github.com/Sitecore/jss/issues/1714

**Comment by illiakovalenko** [comment]
@spike1292 Thank you for submitting your idea! I've added a ticket containing all the details to our internal backlog for thorough review

**Reply by Antonytm** [reply]
@spike1292 
Just a side note about my experience with Angular standalone.

We worked with Angular and Astro integration. The main problem that we faced: Sitecore JSS Angular doesn't support standalone components.

Only Angular Placeholder doesn't work inside Standalone components! And the reason, why it doesn't work: is because it is dependent on Components Factory. It is possible to rewrite the way, how Components Factory is managed. And, use this different way inside the new Placeholder. It should be quite a viable solution without deep modification of Sitecore JSS.

Feel free to reach out to me directly about it.
